cs-vmware/src/main/java/io/cloudslang/content/vmware/entities/AsyncProgressUpdater.java

```diff
@@ -0,0 +1,43 @@
+package io.cloudslang.content.vmware.entities;
+
+import com.vmware.vim25.ManagedObjectReference;
+import io.cloudslang.content.vmware.connection.ConnectionResources;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.lang.Thread.sleep;
+
+/**
+ * Created by giloan on 10/5/2016.
+ */
+public class AsyncProgressUpdater extends ProgressUpdater {
+
+    final private Logger logger = LoggerFactory.getLogger(AsyncProgressUpdater.class);
+
+    public AsyncProgressUpdater(long totalNoBytes, ManagedObjectReference httpNfcLease, ConnectionResources connectionResources) {
+        super(totalNoBytes, httpNfcLease, connectionResources);
+    }
+
+    @Override
+    public final synchronized void updateBytesSent(final long bytesSent) throws Exception {
+        this.bytesSent += bytesSent;
+    }
+
+    @Override
+    public void run() {
+        int percentage = 0;
+        try {
+            while (bytesSent != totalNoBytes) {
```

Comment:
it is sometimes clearer when we use conditions that prove an invariant once loop condition is not satisfied like `bytesSent < totalNoBytes`